Partition_2/include/CGAL/Partition_traits_adapter_2.h

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) 2019  Max-Planck-Institute Saarbruecken (Germany).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+// You can redistribute it and/or modify it under the terms of the GNU
+// General Public License as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// Licensees holding a valid commercial license may use this file in
+// accordance with the commercial license agreement provided with the software.
+//
+// This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0+
+// 
+//
+// Author(s)     : Andreas Fabri
+
+#ifndef CGAL_PARTITION_TRAITS_ADAPTER_2_H
+#define CGAL_PARTITION_TRAITS_ADAPTER_2_H
+
+#include <CGAL/license/Partition_2.h>
+
+#include <boost/call_traits.hpp>
+
+#include <CGAL/property_map.h>
+#include <CGAL/polygon_function_objects.h>
+#include <CGAL/Polygon_2.h>
+#include <list>
+
+namespace CGAL {
+
+  
+template <class Base_traits,class PointPropertyMap>
+class Partition_traits_adapter_2 : public Base_traits
+{
+private:
+  typedef typename Base_traits::Kernel                                   Kernel;
+  typedef Partition_traits_adapter_2<Base_traits,PointPropertyMap>       Self;
+  
+  PointPropertyMap ppmap;
+public:
+
+  Partition_traits_adapter_2(Base_traits base=Base_traits())
+    : Base_traits(base)
+  {}
+  
+  Partition_traits_adapter_2(const PointPropertyMap& ppmap,Base_traits base=Base_traits())
+  : Base_traits(base),ppmap(ppmap)
+  {}
+  
+  typedef typename Kernel::FT FT;
+  typedef typename boost::property_traits<PointPropertyMap>::key_type Point_2;
+  typedef typename boost::call_traits<Point_2>::param_type Arg_type;
+
+  typedef ::std::list<Point_2>                      Container;
```

Comment:
Shouldn't this be the container type of the base traits class?

Reply:
Shall we make it another issue to get rid of the `typedef ... Polygon_2` in the traits classes, so that any model of `Kernel` can be used instead of `Partition_traits_2`?

Reply:
ping @MaelRL

Reply:
It's something to be investigated in another issue/PR, but the Container type is still kinda wrong here.

Reply:
Shall I replace it with `typedef typename Base_traits::Container Container;` and the same for Polygon?

Reply:
I think that would be the best with the current layout, yes.